File: backend/internal/models/comment.go
package models

import "time"

type Comment struct {
	ID          uint64     `db:"id" json:"id"`
	PostID      uint64     `db:"post_id" json:"post_id"`
	UserID      uint64     `db:"user_id" json:"user_id"`
	DisplayName string     `db:"display_name" json:"display_name"`
	Content     string     `db:"content" json:"content"`
	CreatedAt   time.Time  `db:"created_at" json:"created_at"`
	Reactions   []Reaction `db:"-" json:"reactions"`
}

File: frontend/src/types/comment.ts
export interface Reaction {
  emoji: string
  count: number
  reacted_by_me: boolean
}

export interface Comment {
  id: number
  post_id: number
  user_id: number
  display_name: string
  content: string
  created_at: string
  reactions: Reaction[]
}
